cmd/korg/utils.go

```diff
@@ -0,0 +1,183 @@
+	for _, e := range slice {
+		if key == e {
+			return true
+		}
+	}
+
+	return false
+}
+
+func validateOrgs(orgs []string) (valid []string, invalid []string, invalidPresent bool) {
+	valid = []string{}
+	invalid = []string{}
+	invalidPresent = false
+
+	for _, org := range orgs {
+		if !stringInSlice(validOrgs, org) {
+			invalid = append(invalid, org)
+		} else {
+			valid = append(valid, org)
+		}
+	}
+
+	if len(invalid) > 0 {
+		invalidPresent = true
+	}
+
+	return
+}
+
+func readConfig(path string) (*org.Config, error) {
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file at %s: %s", path, err)
+	}
+
+	config := org.Config{}
+	if err := yaml.Unmarshal(contents, &config); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal config from %s: %s", path, err)
+	}
+
+	return &config, nil
+}
+
+func saveConfig(path string, config *org.Config) error {
+	b, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("unable to marshal config: %s", err)
+	}
+
+	info, err := os.Stat(path)
+	if err != nil {
+		return fmt.Errorf("unable to fetch info for %s: %s", path, err)
+	}
+
+	if err := os.WriteFile(path, b, info.Mode()); err != nil {
+		return fmt.Errorf("unable to write to %s: %s", path, err)
+	}
+	return nil
+}
+
+func commitChanges(repoRoot string, configsModified []string, message string) error {
+	r, err := git.PlainOpen(repoRoot)
+	if err != nil {
+		return fmt.Errorf("unable to open repository: %s", err)
+	}
+
+	w, err := r.Worktree()
+	if err != nil {
+		return fmt.Errorf("unable to fetch worktree: %s", err)
+	}
+
+	for _, configModified := range configsModified {
+		_, err := w.Add(configModified)
+		if err != nil {
+			return fmt.Errorf("unable to stage changes: %s", err)
+		}
+	}
+
+	commit, err := w.Commit(message, &git.CommitOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to commit changes: %s", err)
+	}
+
+	_, err = r.CommitObject(commit)
+	if err != nil {
+		return fmt.Errorf("unable to write commit object to repo: %s", err)
+	}
+
+	return nil
+}
+
+// this is an custom implementation which sorts a string slice
+// with a case agnostic heuristic. the default sorting algorithm
+// in Go doesn't ignore case, whereas we do.
+type caseAgnostic []string
+
+func (s caseAgnostic) Len() int {
+	return len(s)
+}
+
+func (s caseAgnostic) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func (s caseAgnostic) Less(i, j int) bool {
+	li := strings.ToLower(s[i])
+	lj := strings.ToLower(s[j])
+
+	return sort.StringsAreSorted([]string{li, lj})
+}
+
+func caseAgnosticSort(arr []string) {
+	sort.Sort(caseAgnostic(arr))
+}
+
+func parseTeam(team string) (string, string, string, error) {
+	parts := strings.Split(team, "/")
+	if len(parts) < 2 || len(parts) > 3 {
+		return "", "", "", fmt.Errorf("invalid team: %s", team)
+	}
+
+	org := parts[0]
+	if !stringInSlice(validOrgs, org) {
+		return "", "", "", fmt.Errorf("invalid team: %s", team)
```

Comment:
typo...invalid org?

Reply:
Removed the team related implementation for now.